### ISSUE TITLE
feat: Rewinding current history without changing others

### ISSUE THREAD

**Issue by beauby** [driftyco/ionic#3043]
**Type**: <span ionic-type>feat</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>Consider the following case: I have a tabbed application, with one tab (```tab1```) dedicated to creating a resource in multiple steps, and an other tab (```tab2```) dedicated to showing a list of resources, and details about a resource. Once the resource has been sent to the server (let's say after step 3), I need to go to ```tab2``` and show the created resource (or at least the resource list). However, when returning to ```tab1```, I need the navigation to be back to the first step of creating a (new) resource.

Hence, I need some kind of history rewinding, but only for the history of ```tab1```. Currently, neither ```clearHistory``` (which clears *all* histories, and does not rewind), nor ```goToHistoryRoot``` (which actually transitions to the history root) do the trick. Moreover, I feel like there should be some kind of manual history management so that applications could wake up in a specific state, with correct histories. I have some ideas as how to build such a functionality, but I'd like to have your thoughts on the matter before making a PR.</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @beauby!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=3043). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=3043)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by beauby** [comment]
While I'm at it: https://github.com/driftyco/ionic/blob/master/js/angular/service/history.js#L137 this method (```setNavViews```) seems to replace the ```currentView``` with a given view (that is already in the ```viewHistory```, since it is references by ```viewId```). The name is rather misleading. What's the rationale?

**Comment by beauby** [comment]
Basically, it would be nice to have a method ```setHistory(historyId, ['state1', 'state2', 'state3'])``` (even nicer if it could handle state parameters), and maybe even a wrapper ```setHistories({historyId1: ['state1', 'state2'], historyId2: ['state3', 'state4', 'state5']})```. What do you guys think?

**Comment by Ionitron** [comment]
Greetings @beauby!

Thank you for the feature request. The humans and I will review this further and possibly add it to our [Feature Request](https://trello.com/b/nNk2Yq1k/ionic-framework) board, if we feel it is appropriate for Ionic.

Thank you for allowing me to assist you.
